File: src/fmp_app/lib/data/models/shipment/create_shipment_request.dart
import '../../../presentation/sender/models/shipment_draft.dart';

class CreateShipmentRequest {
  final String senderPhone;
  final String receiverPhone;

  final String cargoType;
  final String cargoDescription;
  final double cargoWeightKg;

  final bool requiresRefrigeration;
  final bool requiresInsurance;
  final bool isUrgent;

  CreateShipmentRequest({
    required this.senderPhone,
    required this.receiverPhone,
    required this.cargoType,
    required this.cargoDescription,
    required this.cargoWeightKg,
    required this.requiresRefrigeration,
    required this.requiresInsurance,
    required this.isUrgent,
  });

  Map<String, dynamic> toJson() {
    return {
      "senderPhone": senderPhone,
      "receiverPhone": receiverPhone,
      "cargoType": cargoType,
      "cargoDescription": cargoDescription,
      "cargoWeightKg": cargoWeightKg,
      "requiresRefrigeration": requiresRefrigeration,
      "requiresInsurance": requiresInsurance,
      "isUrgent": isUrgent,
    };
  }
}

// 👇 EXTENSION MUST BE OUTSIDE THE CLASS
extension ShipmentDraftMapper on ShipmentDraft {
  CreateShipmentRequest toRequest(String senderPhone) {
    return CreateShipmentRequest(
      senderPhone: senderPhone,
      receiverPhone: receiverPhone ?? "",
      cargoType: cargoType ?? "",
      cargoDescription: cargoDescription ?? "",
      cargoWeightKg: cargoWeightKg ?? 0,
      requiresRefrigeration: requiresRefrigeration,
      requiresInsurance: requiresInsurance,
      isUrgent: isUrgent,
    );
  }
}

File: src/FmpBackend/Dtos/CreateShipmentRequest.cs
namespace FmpBackend.Dtos;
public class CreateShipmentRequest
{
    public string SenderPhone { get; set; } = null!;
    public string ReceiverPhone { get; set; } = null!;

    public string CargoType { get; set; } = default!;
    public string CargoDescription { get; set; } = default!;
    public decimal CargoWeightKg { get; set; }
    public decimal? CargoVolumeCubicMeters { get; set; }
    public int? PackageCount { get; set; }

    public bool RequiresRefrigeration { get; set; }
    public bool RequiresInsurance { get; set; }
    public string? SpecialHandlingInstructions { get; set; }

    public DateTime? PreferredPickupDate { get; set; }
    public DateTime? PreferredDeliveryDate { get; set; }
    public bool IsUrgent { get; set; }

    public decimal? AgreedPrice { get; set; }
    public string? PricePerUnit { get; set; }

    public decimal LoadingCharges { get; set; }
    public decimal UnloadingCharges { get; set; }
    public decimal OtherCharges { get; set; }

    public string? InvoiceNumber { get; set; }
    public decimal? InvoiceValue { get; set; }
    public string? EwayBillNumber { get; set; }
}
